Source/Tests/AOT/AOTGenerator.cpp

```diff
@@ -0,0 +1,161 @@
+#include <FEXCore/Core/Context.h>
+#include <FEXCore/Utils/LogManager.h>
+
+#include "ELFCodeLoader2.h"
+#include "Linux/Utils/ELFContainer.h"
+
+#include <cstddef>
+#include <set>
+#include <sys/resource.h>
+#include <sys/sysinfo.h>
+#include <thread>
+#include <queue>
+
+namespace FEX::AOT {
+void AOTGenSection(FEXCore::Context::Context *CTX, ELFCodeLoader2::LoadedSection &Section) {
+  // Make sure this section is executable and big enough
+  if (!Section.Executable || Section.Size < 16)
+    return;
+
+  std::set<uintptr_t> InitialBranchTargets;
+
+  // Load the ELF again with symbol parsing this time
+  ELFLoader::ELFContainer container{Section.Filename, "", true};
+
+  // Add symbols to the branch targets list
+  container.AddSymbols([&](ELFLoader::ELFSymbol* sym) {
+    auto Destination = sym->Address + Section.ElfBase;
+
+    if (! (Destination >= Section.Base && Destination <= (Section.Base + Section.Size)) ) {
+      return; // outside of current section, unlikely to be real code
+    }
+
+    InitialBranchTargets.insert(Destination);
+  });
+
+  LogMan::Msg::IFmt("Symbol seed: {}", InitialBranchTargets.size());
+
+  // Add unwind entries to the branch target list
+  container.AddUnwindEntries([&](uintptr_t Entry) {
+    auto Destination = Entry + Section.ElfBase;
+
+    if (! (Destination >= Section.Base && Destination <= (Section.Base + Section.Size)) ) {
```

Comment:
This is duplicated a couple of times throughout this function. Maybe move to a lambda?